### DISCUSSION TITLE
4.1.5 release?

### DISCUSSION THREAD

**Discussion by lfcnassif** [sepinf-inc/IPED#1915] [General]
All DEVs,

I'm rolling a 4.1.5 release because of a critical security vulnerability in Google libwebp dependency (#1903). Current release notes are:
https://github.com/sepinf-inc/IPED/blob/4.1.x/ReleaseNotes.txt

Should we include something else?

**Comment by wladimirleite** [comment]
@lfcnassif, can you hold the release a bit? 
I found a small side effect of #1880.
I will submit a PR in the next minutes.

**Reply by lfcnassif** [reply]
Sure, thanks for warning!

**Reply by wladimirleite** [reply]
Submitted PR #1917 describing and fixing this minor issue.

**Reply by wladimirleite** [reply]
Hi @lfcnassif! 
I just found another tricky issue, that would affect 4.1.5.

I noticed in a case processed with the master branch that some WhatsApp messages were duplicated.
It doesn't happen with 4.1.4. It took me quite a while to find the issue, but it ended up being caused by a small typo in this commit https://github.com/sepinf-inc/IPED/pull/1862/commits/1c142f1249eae0e69a9422a89e1653e54b4e4bcb (related to #1843).
As the query to get the chats is broken (there is a missing space), the extractor is using a second attempt, using FQLite.
I will submit a PR fixing the commit mentioned above in a few minutes.

There seems to be another issue, because inside the message extractor there is a mechanism to ignore the duplicated messages. However, for messages with associated media, the mediaHash is used to match the regular and the recovery message, but the mediaHash is always null for the latter. 
I will take a closer look into this later, but it is not related to any changes in recent versions.

**Reply by lfcnassif** [reply]
Ugh... great catch! I executed regression tests and saw more messages were recovered, but didn't notice they were duplicates, thank you for warning and fixing! It was also my fault because I didn't notice the missing space when reviewing... @hauck-jvsh please be careful with this in the future, thank you!

**Reply by hauck-jvsh** [reply]
Sorry, for that, I didn't notice that the query wasn't executing.

**Reply by lfcnassif** [reply]
I think the exception is being catched at the parser class level, to try to recover chats/messages from corrupted databases. In a normal workflow, parsing exceptions are propagated/thrown to the application, are counted and StringsParser is called. That's why the number of parsing exceptions didn't change (I checked in my regression tests) and I thought the new messages were good. Not sure how to warn developers/testers about future situations like this...

**Comment by wladimirleite** [comment]
> I think the exception is being catched at the parser class level, to try to recover chats/messages from corrupted databases.

Exactly.
The current solution seems fine. There is a message in the log about "recovering a corrupted database".
Maybe also printing the stack trace in this point would help developers spotting a similar issue.

**Reply by hauck-jvsh** [reply]
I think that printing this message in the console should help too.

**Comment by lfcnassif** [comment]
I'm running again the WhatsApp regression tests. Already executed a basic processing, report creation and transcription (its code was updated). If the new WA processing looks fine, I'll proceed with 4.1.5 release.

**Reply by lfcnassif** [reply]
All looks fine. Also tested OCR and a few non standard images rendering again.

**Comment by paulobreim** [comment]
iped.exe is 32-bit, right?
Is it possible to have a 64-bit version?

**Reply by wladimirleite** [reply]
The EXE just starts the JVM, which is 64-bit.

**Reply by paulobreim** [reply]
Tks. Is that I see iped.exe with 32 and I thought everything was 32.

![image](https://github.com/sepinf-inc/IPED/assets/381594/ed50ab2b-924a-42d1-9bf6-026b780ebf7e)
